protocol/tx/transaction.go

```diff
@@ -14,6 +14,23 @@ func init() {
 		hash, _ := mapBlockHeader(old)
 		return hash
 	}
+	bc.OutputHash = ComputeOutputID
+}
+
+func ComputeOutputID(sc *bc.SpendCommitment) (h bc.Hash, err error) {
+	defer func() {
+		if r, ok := recover().(error); ok {
+			err = r
+		}
+	}()
+	o := newOutput(valueSource{
+		Ref:      sc.SourceID,
+		Value:    sc.AssetAmount,
+		Position: sc.SourcePosition,
+	}, program{VMVersion: sc.VMVersion, Code: sc.ControlProgram}, sc.RefDataHash, 0)
+
+	h = entryID(o)
+	return h, nil
```

Comment:
That's also true, but note that defer runs after return even when we're not in a panic. Along the lines of @erykwalder's example: https://play.golang.org/p/_qBWbRSA2x